### PR TITLE
Do not unnecessarily fetch the PageRoute twice

### PR DESCRIPTION
**Pull request by aschempp**
Preparing feedback for https://github.com/contao/contao/pull/3605 I noticed these changes would prevent the route object from being loaded from the page registry multiple times.

### CODE REVIEW

File: core-bundle/src/Routing/RouteProvider.php [hunk 1]
```diff
@@ -206,7 +203,7 @@ private function addRoutesForRootPage(PageModel $page, array &$routes): void
             $route->getMethods()
         );
 
-        if (!$urlPrefix || (!$this->legacyRouting && $page->disableLanguageRedirect)) {
+        if (!$urlPrefix || (!$this->legacyRouting && $page->loadDetails() && $page->disableLanguageRedirect)) {
```
Comment (fritzmg) [inline core-bundle/src/Routing/RouteProvider.php]: `PageModel::loadDetails()` does not return a boolean, not sure if we should put that in there?

Comment (aschempp) [reply core-bundle/src/Routing/RouteProvider.php]: it returns the page itself, which "equals" to true. It's only meant to inline-load the details before `disableLanguageRedirect` is queried

Comment (ausi) [reply core-bundle/src/Routing/RouteProvider.php]:
```suggestion
        if (!$urlPrefix || (!$this->legacyRouting && $page->loadDetails()->disableLanguageRedirect)) {
```
Maybe something like this makes that more clear?

Comment (aschempp) [reply core-bundle/src/Routing/RouteProvider.php]: good idea, changed in 018aa0e3ee56f2b18139f6eaa876805b744f679c. That also solves the phpstan issue 😆

**Review by fritzmg — APPROVED** [review]

**Review by Toflar — APPROVED** [review]

**Review by leofeyer — APPROVED** [review]

**Pull request merged**